### PR TITLE
Add Auth#requestToken test cases

### PR DESCRIPTION
**Pull request by gokhanbarisaker**
Resolves: RSA8f1, RSA8f2, RSA8f3 and RSA8f4

### CODE REVIEW

File: test/io/ably/test/rest/RestAuth.java [hunk 1]
```diff
@@ -462,6 +465,230 @@ public Object getTokenRequest(TokenParams params) throws AblyException {
...
+	/**
+	 * Verify client does not have a clientId data within token details and published messages, 
+	 * when a token initialized with null client id
+	 */
+	@Test
+	public void auth_clientid_null_success() {
+		try {
+			final TestVars testVars = RestSetup.getTestVars();
+
+			// implement callback, using Ably instance with key
```
Comment (paddybyers) [inline test/io/ably/test/rest/RestAuth.java]: Please use the same comment style as the existing code.

File: test/io/ably/test/rest/RestAuth.java [hunk 2]
```diff
@@ -462,6 +465,230 @@ public Object getTokenRequest(TokenParams params) throws AblyException {
...
+			Channel channel = ably.channels.get("test");
+			channel.publish(messageName, messageData);
+			
+			// Fetch published message
+			PaginatedResult<Message> result = channel.history(null);
+			Message[] messages = result.items();
+			Message publishedMessage = null;
+			Message message;
+			
+			for (int i = 0; i < messages.length; i++) {
```
Comment (paddybyers) [inline test/io/ably/test/rest/RestAuth.java]: Please use the same style as existing code (there are no spaces between `if` and `(`).

**Comment by paddybyers** [comment]
A couple of style issues but otherwise LGTM. However .. we are in the final stages of migrating to a gradle-based build and so this will need updating because the file has moved. Could you rebase to `gradle-build-rebase` or wait until that is merged and then rebase this PR to the updated master?

**Comment by gokhanbarisaker** [comment]
Hi @paddybyers, I will wait for gradle setup. In the meantime, let me know if I can help with anything related to gradle.

**Comment by paddybyers** [comment]
> I will wait for gradle setup

ok, no problem

**Comment by paddybyers** [comment]
> I will wait for gradle setup

The gradle build is now on `master`.

**Comment by mattheworiordan** [comment]
Thanks @gokhanbarisaker this looks good to me.

**Comment by mattheworiordan** [comment]
@gokhanbarisaker this needs to be rebased on master now btw.

**Comment by gokhanbarisaker** [comment]
Since this is the first time I am using `git rebase`, I will need you to check whether I messed up or it looks fine :). 

I rebased `gokhanbarisaker:RSA8f` commits on top of our up-to-date master. There seems to be some commits (belong to @mattheworiordan, @paddybyers and @lmars ) and that has been carried out with `gokhanbarisaker:RSA8f` branch. I am suspecting I did something wrong over there.

Currently, the most important thing for me is moving commits without breaking new project structure. If you guys approve, I will start to fix missing imports as well as above issues that you mentioned on comments.

**Comment by mattheworiordan** [comment]
@gokhanbarisaker I recommend you take a look at the [`git rebase START_SHA_REF END_SHA_REF --onto master`](https://blog.pivotal.io/labs/labs/git-rebase-onto) process to fix this.  You should only have your commits on top of master.

**Comment by paddybyers** [comment]
Since all files have moved as part of the transition to gradle, you might find it simpler just to reapply your changes by hand to the latest version. As @mattheworiordan says, what we want to see is a PR  with only your commits on top of master.

**Comment by gokhanbarisaker** [comment]
I have updated RSA8f branch https://github.com/gokhanbarisaker/ably-java/commits/RSA8f with the requested changes above. I assume pull request is not reflecting changes made on source branch `gokhanbarisaker:RSA8f`, because it has been closed.

Can you try to re-open the pull request and check if up-to-date changes are being reflected and everything is fine?

Also, @mattheworiordan thanks for the `git rebase --onto ...` suggestion. That worked like a charm.

**Comment by mattheworiordan** [comment]
@gokhanbarisaker we can't do a PR from your repo, only you can.  To do this, simply go to  https://github.com/gokhanbarisaker/ably-java/pulls, hit New Pull request, and create a PR from our master branch to your `RSA8f` branch.  We'll review then.  Thanks.

**Comment by gokhanbarisaker** [comment]
due to changes effected branch, I re-created pull request on https://github.com/ably/ably-java/pull/60

**Pull request closed without merging**